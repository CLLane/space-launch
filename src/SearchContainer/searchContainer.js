import React from "react";
import { connect } from "react-redux";
import { SearchCard } from "../Cards/SearchCard";

export const SearchContainer = ({ search }) => {
  let searchResults = search.map(result => <SearchCard result={result} key={result.name}/>);

  return <>{searchResults}</>;
};

export const mapStateToProps = ({ search }) => ({
  search
});

export default connect(mapStateToProps)(SearchContainer);
