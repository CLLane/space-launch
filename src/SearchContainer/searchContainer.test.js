import React from 'react';
import { shallow } from 'enzyme';
import { SearchContainer, mapStateToProps } from "./searchContainer";

describe('SearchContainer', () => {
  let wrapper;
  let mockResult = {
    name: 'Name'
  }
  let mockSearch = [
   { name: 'search result' }
]
  
  beforeEach(() => {
    wrapper = shallow(
      <SearchContainer 
        result={mockResult}
        search={mockSearch}
        key={1}
      />
    )
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
})

describe('mapStateToProps', () => {
  it('should return an array with search results', () => {
    const mockState = {
      search: [{name: 'falcon'}, {name: 'google'}]
    }
    const expected = {
      search: [{ name: "falcon" }, { name: "google" }]
    };
    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  })
})