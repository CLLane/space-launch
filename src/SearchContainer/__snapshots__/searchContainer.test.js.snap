// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer should match a snapshot 1`] = `
<Fragment>
  <SearchCard
    key="search result"
    result={
      Object {
        "name": "search result",
      }
    }
  />
</Fragment>
`;
